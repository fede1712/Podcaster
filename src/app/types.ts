export interface Podcast {
  "im:name": { label: string };
  "im:image": image[];
  summary: { label: string };
  "im:price": { label: string; attributes: { amount: string; currency: string } };
  "im:contentType": { attributes: { term: string; label: string } };
  rights: { label: string };
  title: { label: string };
  link: { attributes: { rel: string; type: string; href: string } };
  id: { label: string; attributes: { "im:id": string } };
  "im:artist": { label: string; attributes: { href: string } };
  category: { attributes: { "im:id": string; term: string; scheme: string; label: string } };
  "im:releaseDate": { label: string; attributes: { label: string } };
}

export type image = { label: string; attributes: { height: string } };

export interface Episode {
  country: string;
  artworkUrl160: string;
  episodeFileExtension: string;
  episodeContentType: string;
  artworkUrl600: string;
  collectionViewUrl: string;
  trackTimeMillis: number;
  shortDescription: string;
  closedCaptioning: string;
  collectionId: number;
  collectionName: string;
  genres: genres[];
  episodeGuid: string;
  description: string;
  releaseDate: string;
  previewUrl: string;
  trackId: number;
  trackName: string;
  feedUrl: string;
  artistIds: [];
  episodeUrl: string;
  artworkUrl60: string;
  contentAdvisoryRating: string;
  trackViewUrl: string;
  kind: string;
  wrapperType: string;
  genreIds: string[];
  primaryGenreName: string;
  currency: string;
  trackCount: number;
  trackExplicitness: string;
  artistName: string;
  collectionCensoredName: string;
  trackCensoredName: string;
  artworkUrl30: string;
  artworkUrl100: string;
  collectionPrice: number;
  trackPrice: number;
  collectionHdPrice: number;
  collectionExplicitness: string;
}

export type genres = { name: string; id: string };
