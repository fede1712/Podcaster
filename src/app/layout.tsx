import type { Metadata } from "next";
import { Inter } from "next/font/google";
import "./globals.css";
import Link from "next/link";
import { Header } from "./ui/header/header";

export const metadata: Metadata = {
  title: "Podcaster",
  description: "Itunes top podcasts",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className="m-8">
        <Header />
        {children}
      </body>
    </html>
  );
}
