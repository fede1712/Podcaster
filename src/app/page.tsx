import { getPodcasts } from "./lib/data";
import { PodcastCard } from "./ui/podcastCard/podcastCard";

export default async function Home() {
  const data = await getPodcasts();
  console.log(data.feed.entry[1]);

  return (
    <div className="grid grid-cols-4 gap-14 m-14">
      {data.feed.entry.map((podcast: any) => {
        return <PodcastCard author={podcast["im:artist"].label} title={podcast["im:name"].label} src={podcast["im:image"][2].label} />;
      })}
    </div>
  );
}
