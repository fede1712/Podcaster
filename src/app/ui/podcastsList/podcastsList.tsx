"use client";

import { useState } from "react";
import { PodcastCard } from "../podcastCard/podcastCard";
import { Filter } from "../filter/filter";
import { Podcast } from "@/app/types";

export const PodcastsList = ({ data }: { data: Podcast[] }) => {
  const [results, setResults] = useState(data);
  const [search, setSearch] = useState("");

  return (
    <>
      <Filter data={data} results={results} setResults={setResults} setSearch={setSearch} search={search} />
      <div className="grid grid-cols-4 gap-14 m-14">
        {results.map((podcast: Podcast) => {
          const correctImg = podcast["im:image"][podcast["im:image"].length - 1];
          return (
            <PodcastCard key={podcast.id.attributes["im:id"]} author={podcast["im:artist"].label} title={podcast["im:name"].label} src={correctImg} />
          );
        })}
      </div>
    </>
  );
};
