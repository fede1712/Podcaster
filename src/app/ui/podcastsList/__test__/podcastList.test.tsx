import React from "react";
import { render, screen } from "@testing-library/react";
import { Episode, Podcast } from "@/app/types";
import { PodcastsList } from "../podcastsList";

describe("Renders PodcastList correctly", () => {
  test("should render PodcastList", () => {
    const podcast: Podcast[] = [
      {
        "im:name": { label: "Test podcast 1" },
        "im:image": [{ label: "http://example.com/artwork.jpg-1" }],
        summary: { label: "Summary podcast 1" },
        title: { label: "Title podcast 1" },
        id: { attributes: { "im:id": "1" } },
        "im:artist": { label: "Artist Podacast 1" },
      },
      {
        "im:name": { label: "Test podcast 2" },
        "im:image": [{ label: "http://example.com/artwork.jpg-2" }],
        summary: { label: "Summary podcast 2" },
        title: { label: "Title podcast 2" },
        id: { attributes: { "im:id": "2" } },
        "im:artist": { label: "Artist Podacast 2" },
      },
      {
        "im:name": { label: "Test podcast 3" },
        "im:image": [{ label: "http://example.com/artwork.jpg-3" }],
        summary: { label: "Summary podcast 3" },
        title: { label: "Title podcast 3" },
        id: { attributes: { "im:id": "3" } },
        "im:artist": { label: "Artist Podacast 3" },
      },
    ];

    render(<PodcastsList data={podcast} />);

    const nameElement = screen.queryByText("TEST PODCAST 1");
    expect(nameElement).not.toBeNull();

    const artistNameElement = screen.queryByText("Author: Artist Podacast 1");
    expect(artistNameElement).not.toBeNull();

    const { asFragment } = render(<PodcastsList data={podcast} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
