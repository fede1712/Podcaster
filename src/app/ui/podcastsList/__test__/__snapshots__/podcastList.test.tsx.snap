// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders PodcastList correctly should render PodcastList 1`] = `
<DocumentFragment>
  <div
    class="flex justify-end m-14"
  >
    <div
      class="bg-cyan-600 rounded-md p-2 m-2"
    >
      <p
        class="text-white"
      >
        3
      </p>
    </div>
    <input
      class="p-2 border rounded-md"
      placeholder="Filter podcast..."
      type="text"
    />
  </div>
  <div
    class="grid grid-cols-1 md:grid-cols-4 gap-14 m-14"
  >
    <a
      href="podcast/1"
    >
      <div
        class="border rounded-md p-4 shadow-xl"
      >
        <div
          class="flex justify-center relative bottom-14"
        >
          <img
            alt="cover"
            class="rounded-full"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-1&w=256&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-1&w=128&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-1&w=256&q=75 2x"
            style="color: transparent;"
            width="100"
          />
        </div>
        <section
          class="flex justify-center"
        >
          <h2
            class="text-center"
          >
            TEST PODCAST 1
          </h2>
        </section>
        <section
          class="flex justify-center"
        >
          <p
            class="font-light text-sm text-center"
          >
            Author: Artist Podacast 1
          </p>
        </section>
      </div>
    </a>
    <a
      href="podcast/2"
    >
      <div
        class="border rounded-md p-4 shadow-xl"
      >
        <div
          class="flex justify-center relative bottom-14"
        >
          <img
            alt="cover"
            class="rounded-full"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-2&w=256&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-2&w=128&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-2&w=256&q=75 2x"
            style="color: transparent;"
            width="100"
          />
        </div>
        <section
          class="flex justify-center"
        >
          <h2
            class="text-center"
          >
            TEST PODCAST 2
          </h2>
        </section>
        <section
          class="flex justify-center"
        >
          <p
            class="font-light text-sm text-center"
          >
            Author: Artist Podacast 2
          </p>
        </section>
      </div>
    </a>
    <a
      href="podcast/3"
    >
      <div
        class="border rounded-md p-4 shadow-xl"
      >
        <div
          class="flex justify-center relative bottom-14"
        >
          <img
            alt="cover"
            class="rounded-full"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-3&w=256&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-3&w=128&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg-3&w=256&q=75 2x"
            style="color: transparent;"
            width="100"
          />
        </div>
        <section
          class="flex justify-center"
        >
          <h2
            class="text-center"
          >
            TEST PODCAST 3
          </h2>
        </section>
        <section
          class="flex justify-center"
        >
          <p
            class="font-light text-sm text-center"
          >
            Author: Artist Podacast 3
          </p>
        </section>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
